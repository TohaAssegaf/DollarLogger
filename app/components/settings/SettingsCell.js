import styles from './styles'
import React from 'react'
import { Text, TouchableHighlight, View } from 'react-native'

export default class SettingsCell extends React.Component {
  render() {
    return (
      <TouchableHighlight onPress={() => this.props.onTap()} underlayColor='#EEE'>
        <View style={styles.settingsCell}>
          <Text>{this.props.text}</Text>
        </View>
      </TouchableHighlight>
    );
  }
}
