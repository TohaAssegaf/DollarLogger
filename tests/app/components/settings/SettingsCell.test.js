import SettingsCell from '/app/components/settings/SettingsCell'
import React from 'react';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const rendered = renderer.create(<SettingsCell text='Test setting'
  />).toJSON();
  expect(rendered).toMatchSnapshot();
});
