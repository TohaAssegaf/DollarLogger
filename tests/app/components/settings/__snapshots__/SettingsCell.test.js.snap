// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "borderBottomColor": "#222222",
        "borderBottomWidth": 0.5,
        "height": 70,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Test setting
    </Text>
  </View>
</View>
`;
